main/tests/server/src/com/google/refine/expr/functions/arrays/ZipTests.java

```diff
@@ -0,0 +1,48 @@
+
+package com.google.refine.expr.functions.arrays;
+
+import org.testng.annotations.Test;
+
+import com.google.refine.expr.ParsingException;
+import com.google.refine.grel.GrelTestBase;
+
+public class ZipTests extends GrelTestBase {
+
+    @Test
+    public void zipArray() throws ParsingException {
+        String[] test = { "zip([1,2,3], ['A','B','C'])", "[[1, A], [2, B], [3, C]]" };
+        parseEval(bindings, test);
```

Comment:
`parseEval` uses `toString` internally, so I don't really trust it for testing this kind of operation. I think it would be better to have your own help which knows how to compare two arrays of arrays to make sure that you're getting the results that you expect.

Reply:
@tfmorris I have fixed the incorrect type conversion and validated it via the application. 

I will check on updating the unit test to do array comparison.

Reply:
@tfmorris Added test cases with support to compare the output array against expected result.